modules/functional-test/src/main/java/com/intuit/wasabi/tests/library/TestBase.java

```diff
@@ -4336,7 +4338,172 @@ public Response getLogout(AccessToken token, int expectedStatus, APIServerConnec
+	}
+
+    /**
+	  * Sends a GET request to get all the applications assigned to user.
+	  * The response must contain {@link HttpStatus#SC_OK}.
+	  * 
+	  * @param user the user whose list of applications we are interested
+	  * @param expectedStatus  the expected HTTP status code 
+	  * @param apiServerConnector the server connector to use
+	  * @return List of application names
+	  */
+    public List<String>  getUserApplications(APIUser user, int expectedStatus, APIServerConnector apiServerConnector) {
+    	apiServerConnector.setUserNameAndPassword(user.userId, user.password);
+		response = apiServerConnector.doGet("authorization/applications");
+		assertReturnCode(response,expectedStatus);
+		List<Map<String, Object>> jsonStrings = response.jsonPath().get("roleList");
+		List<String> applicationList = new ArrayList<>(jsonStrings.size());
+		
+		//here I am only interested in the name of the application
+		for(int i = 0 ; i < jsonStrings.size(); i++)
+		{
+			String appName = response.jsonPath().get("roleList["+i+"].applicationName[0]").toString();
+			applicationList.add(appName);
+		}
+		apiServerConnector.setUserNameAndPassword(appProperties.getProperty("user-name"),appProperties.getProperty("password"));
+		return applicationList;
+    }
+	
+    /**
+	  * Sends a POST request to assign roles to users against applications.
+	  * The response must contain {@link HttpStatus#SC_OK}.
+	  * Sets createNewApplication to {@code true}.
+	  *
+	  * @param roles the list of userRoles
+	  * @return List of assignment status 
+	  */
+    public List<AssignmentStatus> postUserRolePermission(List<UserRole> roles) {
+    	return postUserRolePermission(roles, HttpStatus.SC_OK);
+	}
+
+	
+    /**
+	  * Sends a POST request to assign roles to users against applications.
+	  * The response must contain {@link HttpStatus#SC_OK}.
+	  * Sets createNewApplication to {@code true}.
+	  *
+	  * @param roles 		  the list of userRoles
+	  * @param expectedStatus  the expected HTTP status code 
+	  * @return List of assignment status 
+	  */
+    public List<AssignmentStatus>  postUserRolePermission(List<UserRole> roles, int expectedStatus) {
+    	return postUserRolePermission(roles, expectedStatus, apiServerConnector);
+	}
+
+    /**
+	  * Sends a POST request to assign roles to users against applications.
+	  * The response must contain {@link HttpStatus#SC_OK}.
+	  * <p>
+	  *
+	  * @param experiment         the experiment to POST
+	  * @param expectedStatus     the expected HTTP status code
+	  * @param apiServerConnector the server connector to use
+	  * @return List of assignment status
+	  */
+    public List<AssignmentStatus>  postUserRolePermission(List<UserRole> roles, int expectedStatus, APIServerConnector apiServerConnector) {
+    	//building the payload
+    	StringBuilder stringBuilder = new StringBuilder();
+		stringBuilder.append("{");
+		stringBuilder.append("\"roleList\":");
```

Comment:
yes mansi know its ugly but i dont have any other way like Gson where i can construct the payload.